torch_geometric/explain/algorithm/subgraphX_explainer.py

```diff
@@ -0,0 +1,646 @@
+                ret_list.append(single_label_ret_list)
+        return ret_list
+
+    def write_from_MCTSNode_list(self, MCTSNode_list):
+        """TODO: Complete Docstrings for Readability"""
+        if isinstance(MCTSNode_list[0], MCTSNode):
+            ret_list = [node.info for node in MCTSNode_list]
+        elif isinstance(MCTSNode_list[0][0], MCTSNode):
+            ret_list = []
+            for single_label_MCTSNode_list in MCTSNode_list:
+                single_label_ret_list = [
+                    node.info for node in single_label_MCTSNode_list
+                ]
+                ret_list.append(single_label_ret_list)
+        return ret_list
+
+    def get_reward_func(self, value_func, node_idx=None):
+        """Runs `__call__` on the `reward_method`"""
+        if self.model_config.task_level == ModelTaskLevel.graph:
+            node_idx = None
+
+        return reward_func(
+            reward_method=self.reward_method,
+            value_func=value_func,
+            node_idx=node_idx,
+            local_radius=self.local_radius,
+            sample_num=self.sample_num,
+            subgraph_building_method=self.subgraph_building_method,
+        )
+
+    def get_mcts_class(self, x, edge_index, node_idx: int = None,
+                       score_func: Callable = None):
+        if self.model_config.task_level == ModelTaskLevel.graph:
+            node_idx = None
+
+        return MCTS(
+            x,
+            edge_index,
+            node_idx=node_idx,
+            device=self.device,
+            score_func=score_func,
+            num_hops=self.num_hops,
+            n_rollout=self.rollout,
+            min_atoms=self.min_atoms,
+            c_puct=self.c_puct,
+            expand_atoms=self.expand_atoms,
+            high2low=self.high2low,
+        )
+
+    def explain(
+        self,
+        model: torch.nn.Module,
+        x: Tensor,
+        edge_index: Tensor,
+        label: int,
+        max_nodes: int = 5,
+        node_idx: Optional[int] = None,
+        saved_MCTSInfo_list: Optional[List[List]] = None,
+    ):
+        # get task classification probabilities
+        with torch.no_grad():
+            probs = model(x, edge_index).softmax(-1)
+
+        # load MCTSInfo_list if present
+        if saved_MCTSInfo_list:
+            results = self.read_from_MCTSInfo_list(saved_MCTSInfo_list)
+
+        if self.model_config.task_level == ModelTaskLevel.graph:
+            # Explanation for Graph Classification Task
+            if not saved_MCTSInfo_list:
+                value_func = GnnNetsGC2valueFunc(model, target_class=label)
+                payoff_func = self.get_reward_func(value_func)
+                self.mcts_state_map = self.get_mcts_class(
+                    x, edge_index, score_func=payoff_func)
+                results = self.mcts_state_map.mcts(verbose=self.verbose)
+
+            # l sharply score
+            value_func = GnnNetsGC2valueFunc(model, target_class=label)
+
+        else:
```

Comment:
Use `elif self.model_config.task_level == ModelTaskLevel.node` here, and have the else raise an exception (similar to GNNExplainer logic flow). This way, as new modes (edge prediction) are incorporated, exceptions are immediately raised.